File: HWAppFoundation/src/main/java/com/hw/hwdroid/foundation/app/HBaseViewHolder.java
package com.hw.hwdroid.foundation.app;

import android.support.annotation.NonNull;
import android.view.View;

import butterknife.ButterKnife;

/**
 * Adapter View Holder
 * <p>
 * Created by ChenJ on 16/8/4.
 */
public class HBaseViewHolder {

    private boolean isBindButterKnife;

    public HBaseViewHolder(@NonNull View view) {
        initViews(view);
    }

    protected void initViews(@NonNull View view) {
        if (null != view && !isBindButterKnife) {
            ButterKnife.bind(this, view);
            isBindButterKnife = true;
        }
    }

}

File: HWAppFoundation/src/main/java/com/hw/hwdroid/foundation/app/HBaseViewHolder.kt
package com.hw.hwdroid.foundation.app

import android.view.View

import butterknife.ButterKnife
import butterknife.Unbinder

/**
 * Adapter View Holder
 *
 *
 * Created by ChenJ on 16/8/4.
 */
open class HBaseViewHolder(view: View) {

    private var unbinder: Unbinder? = null

    init {
        initViews(view)
    }

    open protected fun initViews(view: View) {
        unbinder?.let {
            unbinder = ButterKnife.bind(this, view)
        }
    }

}
